logstash-core-event-java/src/main/java/com/logstash/Logger.java

```diff
@@ -0,0 +1,13 @@
+package com.logstash;
+
+// minimalist Logger interface to wire a logger callback in the Event class
+// for now only warn is defined because this is the only method that's required
+// in the Event class.
+// TODO: (colin) generalize this
+
+public interface Logger {
+
+    // TODO: (colin) complete interface beyond warn when needed
+
+    void warn(String message);
```

Comment:
I see your point now related to the other comment above (I was going backward in the comment :P). So yes, I agree, but would prefer to push back that improvement in another PR.